### ISSUE TITLE
`lime display` with include.xml files

### ISSUE THREAD

**Issue by Gama11** [haxelime/lime#1343]
It would be nice to have code completion support for libraries. I think this more or less just comes down to having `lime display` work with `include.xml` files?

**Comment by player-03** [comment]
It's a good thought, but I don't think we can do that. The problem is Haxe's code completion requires a main class, which libraries lack.

The obvious workaround is to give the library a simple/placeholder main class. But at that point, you might as well create a basic project (`Main.hx` plus `project.xml` are enough to get off the ground). Then Lime will attempt to generate code completion for every library imported by `project.xml`.

Basically, this should all just work as-is. If not, there may be something in the library's code that breaks code completion.

**Comment by Gama11** [comment]
> The problem is Haxe's code completion requires a main class, [...].

That's not true.

**Comment by player-03** [comment]
Really? Ok, then in theory this is possible. Not high priority since you can work around it, but we'll keep it in the list.

**Comment by player-03** [comment]
Looking at this a bit closer, `CommandLineTools` calls `findProjectFile()`, which deliberately excludes include.xml. This is, presumably, to avoid building/testing/running a library, but it also prevents code completion.

Fortunately, you can skip `findProjectFile()` if you specify `lime display include.xml <target>`. Other than that, I guess we could carve out an exception using a series of optional arguments.
